analyzers/src/SonarAnalyzer.CSharp/Rules/StringLiteralShouldNotBeDuplicated.cs

```diff
@@ -38,7 +39,8 @@ public sealed class StringLiteralShouldNotBeDuplicated : StringLiteralShouldNotB
         protected override SyntaxKind[] SyntaxKinds { get; } =
         {
             SyntaxKind.ClassDeclaration,
-            SyntaxKind.StructDeclaration
+            SyntaxKind.StructDeclaration,
+            SyntaxKindEx.RecordDeclaration
         };
 
         public StringLiteralShouldNotBeDuplicated() : base(RspecStrings.ResourceManager) { }
```

Comment:
Can we introduce ILanguageFacade to the base class? That would remove this constructor and the GeneratedCodeRecognizer property. Both from the Csharp and the VbNet implementations.